lib/ace/ext/papeeria/papeeria_latex_highlight_rules.coffee

```diff
@@ -101,154 +140,64 @@ define((require, exports, module) ->
-            "storage.type"
-            "lparen.cite"
-            "variable.parameter.cite"
-            "rparen"
+          "storage.type"
+          "lparen.cite"
+          "variable.parameter.cite"
+          "rparen"
         ]
         regex: "(\\\\(?:cite))({)(\\w*)(})"
       }
-
       {
         token: [
           "keyword"
-          "lparen" + addToken
-          "variable.parameter" + addToken
-          "rparen"
-        ]
-        regex: "(\\\\(?:v?ref|cite(?:[^{]*)))(?:({)([^}]*)(}))?"
-      }
-      {
-        token : "string",
-        regex : "\\\\\\[",
-        next  : pushState("math_latex")
-      }
-
-      {
-        token : "string",
-        regex : "\\${1,2}",
-        next  : pushState("math")
-      }
-
-      {
-        token: "storage.type" + addToken
-        regex: "\\\\[a-zA-Z]+"
-      }
-
-      {
-        token: "constant.character.escape" + addToken
-        regex: "\\\\[^a-zA-Z]?"
-      }
-
-      {
-          defaultToken : "text" + addToken
-      }
-
-    ]
-
-
-    beginRule = (text = "\\w*", pushedState = "start") ->
-      return {
-        token: [
-          "storage.type"
           "lparen"
           "variable.parameter"
           "rparen"
         ]
-        regex: "(\\\\(?:begin))({)(" + text + ")(})"
-        next: pushState(pushedState)
-      }
-
-    endRule = (text = "\\w*") ->
-      return {
-        token: [
-          "storage.type"
-          "lparen"
-          "variable.parameter"
-          "rparen"
-        ]
-        regex: "(\\\\(?:end))({)(" + text + ")(})"
-
-        next: popState
+        regex: "(\\\\(?:v?ref|cite(?:[^{]*)))(?:({)([^}]*)(}))?"
       }
-
-
-    # Function for constructing $$ $$ and \[ \] rules
-    # regex -- (String) -- regex for definition end of Equation
-    latexMathModeConstructor = (closingRegex) ->
-        [{
-            token : "comment",
-            regex : "%.*$"
-          }, {
-            token : "string",
-            regex : closingRegex,
-            next  : popState
-          }, {
-            token: "storage.type." + EQUATION_TOKENTYPE
-            regex: "\\\\[a-zA-Z]+"
-          }, {
-            token: "constant.character.escape." + EQUATION_TOKENTYPE
-            regex: "\\\\[^a-zA-Z]?"
-          }, {
-            token : "error." + EQUATION_TOKENTYPE,
-            regex : "^\\s*$",
-            next : popState
-          }, {
-            defaultToken : "string." + EQUATION_TOKENTYPE
-          }
-        ]
-
-
-    specificTokenForContext = {}
-    specificTokenForContext[LIST_STATE] = LIST_TOKENTYPE
-    specificTokenForContext[EQUATION_STATE] = EQUATION_STATE
-
-    @$rules = {}
-    @$rules["start"] = [
-        beginRule(LIST_REGEX, LIST_STATE)
-        beginRule(EQUATION_REGEX, EQUATION_STATE)
-
-        endRule(EQUATION_REGEX)
-        endRule(LIST_REGEX)
-      ]
+      { token: "string", regex: "\\\\\\[", next: pushState(MATH_LATEX_STATE) }
+      { token: "string", regex: "\\${1,2}", next: pushState(MATH_STATE) }
```

Comment:
So the answer is still no, and here's why. There is another regex (generated by `basicRules` function) after this one, which matches this: `\\[^a-zA-Z]?`. As you can see, it matches with `\$`, and it does so earlier, than `\${1,2}`. So whenever there is `\$`, it would be matched with `\\[^a-zA-Z]?` and then consumed. So everything should be okay, if I don't miss anything.